compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1406,6 +1398,21 @@ private void handleDeclaredWithVar(BLangVariable variable) {
         }
     }
 
+    private void handleWildCardBindingVariable(BLangSimpleVariable variable) {
+        if (!variable.name.value.equals(Names.IGNORE.value)) {
+            return;
+        }
+        BLangExpression bindingExp = variable.expr;
+        BType bindingValueType = bindingExp != null && bindingExp.getBType() != null
+                ? bindingExp.getBType() : variable.getBType();
+        if (!types.isAssignable(bindingValueType, symTable.anyType)) {
+            dlog.error(variable.pos, DiagnosticErrorCode.WILD_CARD_BINDING_ONLY_SUPPORTS_TYPE_ANY);
```

Comment:
Shall we check if we can use this same error in scenarios like

```ballerina
[int, int|error] m = [1, error("")];

record {|
    int i;
    error e;
|} n = {
    i: 1,
    e: error("")
};

var [x, _] = m;
var {i, e: _} = n;

public function main() {
    var [x2, _] = m;
    var {i: i2, e: _} = n;

    (int|error)[2] o = [1, 2];
    var [_, y] = o;
}
```

If it is a lot of work, we can do this separately.

Reply:
It was a simple fix. Updated and added the tests